### DISCUSSION TITLE
When sending packets with http 2.0, there is a problem that the host value is continuously transmitted.

### DISCUSSION THREAD

**Discussion by neospider84** [encode/httpx#2781] [Q&A] [answered]
Hello I am korean.

In http version 1, host is the default, but
In http 2 version, authority value is given instead of host.

Is there any way to exclude the host value when sending packets with http2?

**Comment by zanieb** [comment]
Hi @neospider84

I'm going to transfer this to a "Discussion" instead of an issue since it's a question.

Could you provide an example that demonstrates what's happening and what you're looking for?

**Comment by neospider84** [comment]
Thank you for answer.
For example if I send a packet from python to httpx
Even if http2.0 is used, the fiddler header value uses the host value, not the authority.
I wanted to exclude it because http2.0 doesn't use host header, but I couldn't figure out how.

Is there a way to exclude the host when sending packets with the http2 version using httpx???

**Reply by zanieb** [reply]
Could you share a code example, the full request you see in Fiddler (or using something like https://public.requestbin.com/r/), and the request you want instead?

**Reply by neospider84** [reply]
https://m.search.naver.com/search.naver?sm=mtp_hty.top&where=m&query=%EC%A0%84%EC%9E%90%EB%8B%B4%EB%B0%B0%EC%9E%85%ED%98%B8%ED%9D%A1%EC%95%A1%EC%83%81
리퀘스트 헤더 Headers({'host': 'm.search.naver.com', 'connection': 'keep-alive', 'sec-ch-ua': '"Chromium";v="112", "Google Chrome";v="112", "Not:A-Brand";v="99"'.........................other omitted

***The above values are exported from python to httpx.

Key: :method; Value: GET
Key: :authority; Value: m.search.naver.com
Key: :scheme; Value: https
Key: :path; Value: /search.naver?sm=mtb_hty.top&where=m&oquery=%EA%B0%90%EC%9E%90&tqi=i7ov9dprfjCssnB5gbRssssstkG-140934&query=%EA%B0%90%EC%9E%90
Key: sec-ch-ua; Value: "Not.A/Brand";v="8", "Chromium";v="114", "Google Chrome";v="114"...........other omitted

***The above value is the packet value transmitted from the actual smartphone.

the difference between the two is
httpx-host: m.search.naver.com
Smartphone - authority : m.search.naver.com

Adding an authority value is currently possible.

Issues are automatically sent

host & connection.

Is it possible to set the host & connection not to be automatically transmitted?
A specific site is being blocked by judging this as abnormal traffic.

**Reply by lovelydinosaur** [reply, marked as answer]
> Is there any way to exclude the host value when sending packets with http2?

You don't need to.

The request will automatically use either an `:authority` header for HTTP/2 or a `Host` header for HTTP/1.1.
